cmd/aks-periscope/aks-periscope.go

```diff
@@ -120,36 +260,31 @@ func main() {
 			}
 
 			log.Printf("Diagnoser: %s, export data", d.GetName())
-			if err = exp.Export(d); err != nil {
+			if err = runExportersForDataProducer(exporters, d); err != nil {
 				log.Printf("Diagnoser: %s, export data failed: %v", d.GetName(), err)
 			}
 		}(d)
 	}
+}
 
-	diagnoserGrp.Wait()
-
-	zip, err := exporter.Zip(dataProducers)
-	if err != nil {
-		log.Printf("Could not zip data: %v", err)
-	} else {
-		if err := exp.ExportReader(hostname+".zip", bytes.NewReader(zip.Bytes())); err != nil {
-			log.Printf("Could not export zip archive: %v", err)
+// runExporters run the exporters for a data producer
+func runExportersForDataProducer(exporters []interfaces.Exporter, dataProducer interfaces.DataProducer) error {
+	var result error
+	for _, e := range exporters {
+		if err := e.Export(dataProducer); err != nil {
+			result = multierror.Append(result, err)
 		}
 	}
-
-	// TODO: Hack: for now AKS-Periscope is running as a deamonset so it shall not stop (or the pod will be restarted)
-	// Revert from https://github.com/Azure/aks-periscope/blob/b98d66a238e942158ef2628a9315b58937ff9c8f/cmd/aks-periscope/aks-periscope.go#L70
-	select {}
-
-	// TODO: remove this //nolint comment once the select{} has been removed
-	//nolint:govet
+	return result
 }
 
-func contains(flagsList []string, flag string) bool {
```

Comment:
💡Key: Not sure what are these changes, and what was the intention of removing this function?